### PR TITLE
New resource aws_bedrock_inference_profile

### LINKED ISSUE #40288: [New Resource]: "aws_bedrock_inference_profile" - Cost allocation and metrics for bedrock
### Description

> Application Inference profiles - These are inference profiles created by users (user defined). This helps our customers to track costs and model usage. You can create an inference profile that routes model invocation requests to one region (with foundation model identifier) or to multiple regions (by using cross region inference profile identifier).

### Requested Resource(s) and/or Data Source(s)

* aws_bedrock_inference_profile

### Potential Terraform Configuration

```terraform
resource "aws_bedrock_inference_profile" "test" {
  name             = "Claude 3 haiku for p123"
  description      = "Tracks costs for project 123"

  model_source {
    copy_from = "arn:aws:bedrock:eu-central-1::inference-profile/eu.anthropic.claude-3-haiku-20240307-v1:0"
  }

  tags = {
    Name = "test"
    Project = "123"
 }
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2024/11/amazon-bedrock-cost-allocation-tags-inference-profiles/
* https://community.aws/content/2oa6QAE7tgxmAsgAExXY2B2MqCV/getting-started-with-bedrock-application-inference-profile

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by chlunde**
### Description

Introduce new resource aws_bedrock_inference_profile, which allows cost tracking and metrics.

### Relations

Closes #40288

### References
* https://aws.amazon.com/about-aws/whats-new/2024/11/amazon-bedrock-cost-allocation-tags-inference-profiles/
* https://community.aws/content/2oa6QAE7tgxmAsgAExXY2B2MqCV/getting-started-with-bedrock-application-inference-profile

### Output from Acceptance Testing
```console
make testacc TESTS=TestAccBedrockInferenceProfile_.* PKG=bedrock
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/bedrock/... -v -count 1 -parallel 20 -run='TestAccBedrockInferenceProfile_.*'  -timeout 360m
2024/11/26 13:08:22 Initializing Terraform AWS Provider...
=== RUN   TestAccBedrockInferenceProfile_basic
=== PAUSE TestAccBedrockInferenceProfile_basic
=== RUN   TestAccBedrockInferenceProfile_tags
=== PAUSE TestAccBedrockInferenceProfile_tags
=== RUN   TestAccBedrockInferenceProfile_disappears
=== PAUSE TestAccBedrockInferenceProfile_disappears
=== CONT  TestAccBedrockInferenceProfile_basic
=== CONT  TestAccBedrockInferenceProfile_disappears
=== CONT  TestAccBedrockInferenceProfile_tags
--- PASS: TestAccBedrockInferenceProfile_disappears (31.94s)
--- PASS: TestAccBedrockInferenceProfile_basic (37.83s)
--- PASS: TestAccBedrockInferenceProfile_tags (46.07s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/bedrock	46.205s
```

### CODE REVIEW

**Review by gdavison — COMMENTED** [review]
Thanks for the PR, @chlunde!

The PR #40305 will add tagging tests for `bedrock` resources.

It is ok to share code between the resource and data source in some cases. The schemas should be separate, but other code can be shared

File: internal/service/bedrock/inference_profile.go [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceInferenceProfile) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// There is no update function for this resource, but tags can be updated
+}
```
[the remark this answers is not in the document]
Comment (chlunde) [reply internal/service/bedrock/inference_profile.go]: Thanks! I also had to add `UseStateForUnknown`

**Review by gdavison — COMMENTED** [review]
Thanks for the PR, @chlunde. I've made a few changes:

* Removed optional attributes from the `basic` test
* Added a separate test for `description`
* Generated tagging tests

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccBedrockInferenceProfile_' PKG=bedrock
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/bedrock/... -v -count 1 -parallel 20  -run=TestAccBedrockInferenceProfile_ -timeout 360m
2024/12/04 16:10:34 Initializing Terraform AWS Provider...
=== RUN   TestAccBedrockInferenceProfile_tags
=== PAUSE TestAccBedrockInferenceProfile_tags
=== RUN   TestAccBedrockInferenceProfile_tags_null
=== PAUSE TestAccBedrockInferenceProfile_tags_null
=== RUN   TestAccBedrockInferenceProfile_tags_EmptyMap
=== PAUSE TestAccBedrockInferenceProfile_tags_EmptyMap
=== RUN   TestAccBedrockInferenceProfile_tags_AddOnUpdate
=== PAUSE TestAccBedrockInferenceProfile_tags_AddOnUpdate
=== RUN   TestAccBedrockInferenceProfile_tags_EmptyTag_OnCreate
=== PAUSE TestAccBedrockInferenceProfile_tags_EmptyTag_OnCreate
=== RUN   TestAccBedrockInferenceProfile_tags_EmptyTag_OnUpdate_Add
=== PAUSE TestAccBedrockInferenceProfile_tags_EmptyTag_OnUpdate_Add
=== RUN   TestAccBedrockInferenceProfile_tags_EmptyTag_OnUpdate_Replace
=== PAUSE TestAccBedrockInferenceProfile_tags_EmptyTag_OnUpdate_Replace
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_providerOnly
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_providerOnly
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_nonOverlapping
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_nonOverlapping
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_overlapping
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_overlapping
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_emptyResourceTag
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccBedrockInferenceProfile_tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccBedrockInferenceProfile_tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccBedrockInferenceProfile_tags_ComputedTag_OnCreate
=== PAUSE TestAccBedrockInferenceProfile_tags_ComputedTag_OnCreate
=== RUN   TestAccBedrockInferenceProfile_tags_ComputedTag_OnUpdate_Add
=== PAUSE TestAccBedrockInferenceProfile_tags_ComputedTag_OnUpdate_Add
=== RUN   TestAccBedrockInferenceProfile_tags_ComputedTag_OnUpdate_Replace
=== PAUSE TestAccBedrockInferenceProfile_tags_ComputedTag_OnUpdate_Replace
=== RUN   TestAccBedrockInferenceProfile_tags_IgnoreTags_Overlap_DefaultTag
=== PAUSE TestAccBedrockInferenceProfile_tags_IgnoreTags_Overlap_DefaultTag
=== RUN   TestAccBedrockInferenceProfile_tags_IgnoreTags_Overlap_ResourceTag
=== PAUSE TestAccBedrockInferenceProfile_tags_IgnoreTags_Overlap_ResourceTag
=== RUN   TestAccBedrockInferenceProfile_basic
=== PAUSE TestAccBedrockInferenceProfile_basic
=== RUN   TestAccBedrockInferenceProfile_disappears
=== PAUSE TestAccBedrockInferenceProfile_disappears
=== RUN   TestAccBedrockInferenceProfile_description
=== PAUSE TestAccBedrockInferenceProfile_description
=== CONT  TestAccBedrockInferenceProfile_tags
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_emptyResourceTag
=== CONT  TestAccBedrockInferenceProfile_tags_IgnoreTags_Overlap_DefaultTag
=== CONT  TestAccBedrockInferenceProfile_tags_ComputedTag_OnUpdate_Replace
=== CONT  TestAccBedrockInferenceProfile_tags_EmptyTag_OnUpdate_Replace
=== CONT  TestAccBedrockInferenceProfile_disappears
=== CONT  TestAccBedrockInferenceProfile_basic
=== CONT  TestAccBedrockInferenceProfile_tags_IgnoreTags_Overlap_ResourceTag
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_updateToResourceOnly
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_updateToProviderOnly
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_overlapping
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_nonOverlapping
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_providerOnly
=== CONT  TestAccBedrockInferenceProfile_tags_AddOnUpdate
=== CONT  TestAccBedrockInferenceProfile_tags_EmptyTag_OnUpdate_Add
=== CONT  TestAccBedrockInferenceProfile_tags_EmptyTag_OnCreate
=== CONT  TestAccBedrockInferenceProfile_tags_EmptyMap
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_nullNonOverlappingResourceTag
=== CONT  TestAccBedrockInferenceProfile_tags_ComputedTag_OnUpdate_Add
=== CONT  TestAccBedrockInferenceProfile_description
--- PASS: TestAccBedrockInferenceProfile_disappears (17.30s)
=== CONT  TestAccBedrockInferenceProfile_tags_ComputedTag_OnCreate
--- PASS: TestAccBedrockInferenceProfile_basic (18.21s)
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_nullOverlappingResourceTag
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_emptyResourceTag (28.82s)
=== CONT  TestAccBedrockInferenceProfile_tags_null
--- PASS: TestAccBedrockInferenceProfile_tags_EmptyMap (29.09s)
=== CONT  TestAccBedrockInferenceProfile_tags_DefaultTags_emptyProviderOnlyTag
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_updateToProviderOnly (37.67s)
--- PASS: TestAccBedrockInferenceProfile_description (38.60s)
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_nullNonOverlappingResourceTag (40.62s)
--- PASS: TestAccBedrockInferenceProfile_tags_ComputedTag_OnUpdate_Add (42.33s)
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_nullOverlappingResourceTag (25.36s)
--- PASS: TestAccBedrockInferenceProfile_tags_ComputedTag_OnCreate (27.50s)
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_updateToResourceOnly (45.85s)
--- PASS: TestAccBedrockInferenceProfile_tags_EmptyTag_OnUpdate_Replace (45.87s)
--- PASS: TestAccBedrockInferenceProfile_tags_AddOnUpdate (48.84s)
--- PASS: TestAccBedrockInferenceProfile_tags_null (21.52s)
--- PASS: TestAccBedrockInferenceProfile_tags_IgnoreTags_Overlap_DefaultTag (50.44s)
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_emptyProviderOnlyTag (21.51s)
--- PASS: TestAccBedrockInferenceProfile_tags_ComputedTag_OnUpdate_Replace (51.65s)
--- PASS: TestAccBedrockInferenceProfile_tags_IgnoreTags_Overlap_ResourceTag (52.98s)
--- PASS: TestAccBedrockInferenceProfile_tags_EmptyTag_OnUpdate_Add (54.21s)
--- PASS: TestAccBedrockInferenceProfile_tags_EmptyTag_OnCreate (54.21s)
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_overlapping (65.09s)
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_nonOverlapping (65.14s)
--- PASS: TestAccBedrockInferenceProfile_tags_DefaultTags_providerOnly (72.11s)
--- PASS: TestAccBedrockInferenceProfile_tags (76.43s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/bedrock	82.058s
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Comment by chlunde** [comment]
@gdavison @ewbankkit thanks!

**Pull request merged**